src/index/base.cpp

```diff
@@ -380,7 +380,7 @@ bool BaseIndex::BlockUntilSyncedToCurrentChain() const
     }
 
     LogPrintf("%s: %s is catching up on block notifications\n", __func__, GetName());
-    SyncWithValidationInterfaceQueue();
+    m_chain->context()->validation_signals->SyncWithValidationInterfaceQueue();
```

Comment:
See the comments [here](https://github.com/bitcoin/bitcoin/pull/28960#discussion_r1455548750) for the reasoning. Basically we want to guard in the shutdown (`Stop`) case and it is less important to do so in the rest, since we have tests heavily exercising those paths. Question is also on which de-reference we should be guarding and what to do about the existing unguarded lines that call `m_chain` and its members.